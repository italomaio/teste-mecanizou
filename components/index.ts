export { default as Card } from "./Card/Card";
export { default as RootProviders } from "./RootProviders/RootProviders";
export { default as Logo, type LogoProps } from "./Logo/Logo";
export { default as Input } from "./Input/Input";
export { default as Label, type LabelProps } from "./Label/Label";
export {
  default as Button,
  type ButtonVariant,
  type ButtonSize,
  type ButtonProps,
} from "./Button/Button";
export { default as ThemeToggle } from "./ThemeToggle/ThemeToggle";
export { default as ProductLink } from "./ProductLink/ProductLink";
export { default as Navbar } from "./Navbar/Navbar";
export { default as Skeleton } from "./Skeleton/Skeleton";
export { default as Pagination } from "./Pagination/Pagination";
export { default as ProductRating } from "./ProductRating/ProductRating";
export { default as ProductCard } from "./ProductCard/ProductCard";
export { default as ProductQuantity } from "./ProductQuantity/ProductQuantity";
export { default as ProductImages } from "./ProductImages/ProductImages";
export { default as ProductReviews } from "./ProductReviews/ProductReviews";
export { default as Stars } from "./ProductRating/Stars";
export { default as Loader } from "./Loader/Loader";
export { default as ProductPrices } from "./ProductPrices/ProductPrices";
export {
  default as UserDropdown,
  type UserDropdownProps,
} from "./UserDropdown/UserDropdown";
