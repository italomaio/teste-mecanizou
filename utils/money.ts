export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
