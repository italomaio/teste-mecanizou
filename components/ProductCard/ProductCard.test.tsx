import { fireEvent, render, screen } from "@/tests/utils";
import { ProductCard } from "@/components";
import { useRouter } from "next/navigation";

const product = {
  id: 1,
  title: "Essence Mascara Lash Princess",
  description:
    "The Essence Mascara Lash Princess is a popular mascara known for its volumizing and lengthening effects. Achieve dramatic lashes with this long-lasting and cruelty-free formula.",
  category: "beauty",
  price: 9.99,
  discountPercentage: 10.48,
  rating: 2.56,
  stock: 99,
  tags: ["beauty", "mascara"],
  brand: "Essence",
  sku: "BEA-ESS-ESS-001",
  weight: 4,
  dimensions: {
    width: 15.14,
    height: 13.08,
    depth: 22.99,
  },
  warrantyInformation: "1 week warranty",
  shippingInformation: "Ships in 3-5 business days",
  availabilityStatus: "In Stock",
  reviews: [
    {
      rating: 3,
      comment: "Would not recommend!",
      date: "2025-04-30T09:41:02.053Z",
      reviewerName: "Eleanor Collins",
      reviewerEmail: "eleanor.collins@x.dummyjson.com",
    },
    {
      rating: 4,
      comment: "Very satisfied!",
      date: "2025-04-30T09:41:02.053Z",
      reviewerName: "Lucas Gordon",
      reviewerEmail: "lucas.gordon@x.dummyjson.com",
    },
    {
      rating: 5,
      comment: "Highly impressed!",
      date: "2025-04-30T09:41:02.053Z",
      reviewerName: "Eleanor Collins",
      reviewerEmail: "eleanor.collins@x.dummyjson.com",
    },
  ],
  returnPolicy: "No return policy",
  minimumOrderQuantity: 48,
  meta: {
    createdAt: "2025-04-30T09:41:02.053Z",
    updatedAt: "2025-04-30T09:41:02.053Z",
    barcode: "5784719087687",
    qrCode: "https://cdn.dummyjson.com/public/qr-code.png",
  },
  images: [
    "https://cdn.dummyjson.com/product-images/beauty/essence-mascara-lash-princess/1.webp",
  ],
  thumbnail:
    "https://cdn.dummyjson.com/product-images/beauty/essence-mascara-lash-princess/thumbnail.webp",
};

describe("ProductCard", () => {
  const routerPushMock = jest.fn();

  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({
      push: routerPushMock,
    });
  });

  it("Should render correctly", () => {
    const { container } = render(<ProductCard product={product} />);
    expect(container).toBeInTheDocument();
  });

  it("Should redirect on click", () => {
    render(<ProductCard product={product} />);
    fireEvent.click(screen.getByRole("button", { name: /Buy/i }));
    expect(routerPushMock).toHaveBeenCalled();
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(<ProductCard product={product} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
