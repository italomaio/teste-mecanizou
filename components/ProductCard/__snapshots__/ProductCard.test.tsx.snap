// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductCard Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white shadow-md rounded-md p-4 dark:bg-zinc-800 dark:text-white dark:border dark:border-zinc-700 dark:shadow-none"
  >
    <article>
      <figure
        class="aspect-square flex items-center"
      >
        <a
          aria-label="View product 1"
          class="focus-visible-ring flex-col flex w-full justify-center items-center relative aspect-[16/9]"
          href="/products/1"
        >
          <img
            alt="Thumbnail for Essence Mascara Lash Princess"
            class="object-contain"
            src="https://cdn.dummyjson.com/product-images/beauty/essence-mascara-lash-princess/thumbnail.webp"
          />
        </a>
      </figure>
      <header
        class="text-left flex flex-col justify-start"
      >
        <a
          aria-label="View product 1"
          class="focus-visible-ring flex flex-col"
          href="/products/1"
        >
          <label
            class="focus-visible-ring p-0 text-left font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white block text-md m-0 cursor-pointer hover:underline lg:min-h-auto leading-tight"
          >
            Essence Mascara Lash Princess
          </label>
        </a>
        <div
          class="flex items-center gap-[2px] py-2"
        >
          <div
            class="flex flex-row gap-[3px]"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star dark:text-yellow-300 text-yellow-500"
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star dark:text-yellow-300 text-yellow-500"
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star dark:text-yellow-300 text-yellow-500"
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star dark:text-white text-gray-700"
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star dark:text-white text-gray-700"
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
          </div>
          <a
            aria-label="View product 1"
            class="focus-visible-ring flex flex-col"
            href="/products/1"
          >
            <label
              class="inline-block focus-visible-ring p-0 text-left font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white text-xs m-0 px-1 cursor-pointer hover:underline"
            >
              3 reviews
            </label>
          </a>
        </div>
      </header>
      <div
        class="flex flex-col items-start py-4 content-end"
      >
        <div
          class="flex flex-col py-2"
        >
          <label
            class="inline-block focus-visible-ring p-0 text-left font-medium text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-500 dark:text-zinc-500 line-through italic mb-0"
          >
            R$ 9,99
          </label>
          <label
            class="inline-block focus-visible-ring p-0 mb-2 text-left peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-3xl font-bold dark:text-hot-cinnamon text-sapphire"
          >
            R$ 8,94
          </label>
        </div>
        <button
          aria-label="Buy Essence Mascara Lash Princess"
          class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed bg-hot-cinnamon text-white hover:bg-sapphire px-4 py-2 text-base w-full"
        >
          Comprar
        </button>
      </div>
    </article>
  </div>
</DocumentFragment>
`;
